src/components/ListingComponent/ListingComponent.vue

```diff
@@ -8,13 +9,36 @@ import { ref, watch } from "vue";
 
 // props
 const props = defineProps<{
+  depositList: (Event | ValidDeposit)[];
```

Comment:
Opa, acho que o depositList agora só pode ser passado como ValidDeposit, correto? Se esse for o caso, seria interessante colocá-lo para aceitar somente essa tipagem.